### PR TITLE
adds method_missing support to BunnyMock::Bunny

### PR DESCRIPTION
**Pull request by mpelzsherman**
Needed to support this story:

https://www.pivotaltracker.com/story/show/74329842

### CODE REVIEW

**Comment by martinstreicher** [comment]
Please read this: http://robots.thoughtbot.com/always-define-respond-to-missing-when-overriding and/or this: http://blog.enriquez.me/2010/2/21/dont-forget-about-respond-to-when-implementing-method-missing/

**Comment by mpelzsherman** [comment]
@martinstreicher - added implementation of `respond_to?`. OK to merge?

File: spec/lib/bunny_mock_spec.rb [hunk 1]
```diff
@@ -61,7 +61,27 @@
 end
 
 describe BunnyMock do
-  let(:bunny) { BunnyMock::Bunny.new }
+  let(:bunny) { BunnyMock::Bunny.new(queue_exists?:true) }
```
Comment (cdemyanovich) [inline spec/lib/bunny_mock_spec.rb:64]: Add a space between `queue_exists?:` and `true`.

File: spec/lib/bunny_mock_spec.rb [hunk 2]
```diff
@@ -61,7 +61,27 @@
...
+
+  describe "#method_missing" do
+    it "returns true for queue_exists?" do
+      expect(bunny.queue_exists?('foo')).to eq(true)
+    end
+  end
+
+  describe "#respond_to?" do
+    it "returns true for queue_exists?" do
+      expect(bunny.respond_to?(:queue_exists?)).to eq(true)
```
Comment (cdemyanovich) [inline spec/lib/bunny_mock_spec.rb:74]: Not a necessity, but consider using the [`respond_to` matcher](https://www.relishapp.com/rspec/rspec-expectations/v/2-14/docs/built-in-matchers/respond-to-matcher).

File: lib/bunny_mock.rb [hunk 3]
```diff
@@ -3,6 +3,24 @@
...
 
   class Bunny
+    attr_accessor :attrs
+
+    def initialize(attrs = {})
+      self.attrs = attrs.dup
+    end
+
+    def method_missing(method, *args)
+      if attrs.has_key? method
```
Comment (martinstreicher) [inline lib/bunny_mock.rb:13]: Could be written as `attrs.fetch method, super`

**Comment by mpelzsherman** [comment]
Looks like this wasn't needed; closing.

**Pull request closed without merging**